### PR TITLE
Search results add error message handling

### PR DESCRIPTION
**Pull request by iisa**
- adds new property: `errorMessage`
- updates demo to use `errorMessage` and flags that are used in archive.org
- style updates:
  - padding
  - larger input size
- method grouping updates
- search results
  - add page number
  - toggle styling between results with & without images

----
To test locally:
- pull branch & run local server with: `npm install && npm start`
- go to: localhost:8000/demo
- search: `man`
  -> spinner comes up
  -> results come back
  -> page number is with the search result
- search: `foooooasdfasdfdasfo`
  -> spinner comes up
  -> at return: `0 results` displays where search results are

test online:
- go to: www-isa.archive.org/details/goody?q=foooooasdfasdfdasfo 
-> search shortcut menu shows up -> click it opens
-> `0 results` shows up in lieu of the results panel
-> search new word: `man`
-> spinner comes up -> it's styled properly
-> results come back:
  -> page number is part of search result
  -> result is flush to left

### CODE REVIEW

File: demo/index.html [hunk 1]
```diff
@@ -199,8 +202,19 @@ <h2>Change book</h2>
...
-      document.addEventListener('BookReader:SearchCallback', () => {
+      document.addEventListener('BookReader:SearchCallback', (event) => {
+        const results = event.detail.results || [];
+        const rawBlob = event.detail.actual || {};
+
+        menuSlider.shadowRoot.querySelector('ia-book-search-results').results = results;
         menuSlider.shadowRoot.querySelector('ia-book-search-results').queryInProgress = false;
+
+        if (rawBlob.error) {
+          console.log('error', rawBlob);
```
Comment (bfalling) [inline demo/index.html:213]: SUGGESTION: Remove debug output. Or if useful to keep, include contextual information, so it's a more "complete" error.

File: src/book-search-result.js [hunk 2]
```diff
@@ -37,11 +37,16 @@ export class BookSearchResult extends LitElement {
...
+    const { par = [] } = match;
+    const [resultDetails = {}] = par;
+    const pageNumber = Number.isInteger(resultDetails.page)
+      ? html`<p class="page-num">Page -${resultDetails.page}-</p>` : nothing;
     const coverImage = html`<img src="${match.cover}" />`;
     return html`
       <li @click=${this.resultSelected}>
         ${match.cover ? coverImage : nothing}
-        <h4>${match.title}</h4>
+        <h4>${match.title ? match.title : nothing}</h4>
```
Comment (bfalling) [inline src/book-search-result.js:48]: SUGGESTION: Would this be simpler as `match.title || nothing`?

**Review by bfalling — APPROVED** [review]
No blockers. A couple questions/suggestions.

File: demo/index.html (hunk 1 above)
Comment (iisa) [reply demo/index.html:213]: updated.

File: src/book-search-result.js (hunk 2 above)
Comment (iisa) [reply src/book-search-result.js:48]: yes!

**Pull request merged**